toolchain/check/pattern_match.cpp

```diff
@@ -151,14 +151,15 @@ auto MatchContext::EmitPatternMatch(Context& context,
     case SemIR::SymbolicBindingPattern::Kind: {
       CARBON_CHECK(kind_ == MatchKind::Callee);
       auto binding_pattern = pattern.inst.As<SemIR::AnyBindingPattern>();
-      auto bind_name = context.insts().GetAs<SemIR::AnyBindName>(
-          binding_pattern.bind_name_id);
+      auto bind_name_id = context.bind_name_cache()
+                              .Lookup(binding_pattern.entity_name_id)
+                              .value();
```

Comment:
Does it make sense to remove the entry from the cache here?

Reply:
I think removing it outright could mask bugs, by allowing it to be re-inserted (which should never happen), but we can at least invalidate the entry. Done.